src/as2fm/scxml_converter/scxml_entries/scxml_base.py

```diff
@@ -28,11 +30,29 @@ def get_tag_name() -> str:
         """Get the tag name of the XML element."""
         raise NotImplementedError
 
-    @staticmethod
-    def from_xml_tree(xml_tree: ET.Element) -> "ScxmlBase":
-        """Create a ScxmlBase object from an XML tree."""
+    @classmethod
+    def from_xml_tree(cls, xml_tree: ET.Element) -> "ScxmlBase":
+        """External interface to create a ScxmlBase object from an XML tree."""
+        instance = cls.from_xml_tree_impl(xml_tree)
+        instance.set_xml_tree(xml_tree)
+        return instance
+
+    @classmethod
+    def from_xml_tree_impl(cls, xml_tree: ET.Element) -> "ScxmlBase":
+        """Child-specific implementation to create a ScxmlBase object from an XML tree."""
         raise NotImplementedError
 
+    def set_xml_tree(self, xml_tree: ET.Element):
+        """Set the xml_element this object was made from"""
+        self.xml_tree = xml_tree
+
+    def get_xml_tree(self) -> Optional[ET.Element]:
```

Comment:
I wonder if we should call this `get_xml_element`, though it might not really be consistent with the `from_xml_tree` method.

Reply:
I changed it to `get_xml_origin` to avoid confusion. What do you think?

Reply:
Technically, the element also contains the tree. So we could also change `from_xml_tree` to `from_xml`?